### PR TITLE
Fixed Flickr doesn't recognise the permission set

### PR DESCRIPTION
**Pull request by brianwozeniak**
Currently when using Flickr it will result in the error: "Oops! Flickr doesn't recognise the permission set." - The authorize URL must have perms specified on it:

https://www.flickr.com/groups/51035612836@N01/discuss/72157692896948105/

Docs are here:

https://www.flickr.com/services/api/auth.oauth.html

I have verified that once this is added everything works correctly again.

### CODE REVIEW

**Comment by atymic** [comment]
"Additionally, you can pass the optional perms= parameter, asking for read, write, or delete privileges. This parameter will override the setting defined in your application's authentication flow."

Sounds like you should specify in your config on their end?

**Comment by brianwozeniak** [comment]
Doesn’t work there, tried that. Feel free to test for yourself if you want.

Brian

**Comment by atymic** [comment]
Just want to make sure we don't break anyone who has write perms set, and this overrides that.

How about we add it a config option, not set by default?

**Comment by brianwozeniak** [comment]
Unfortunately Flickr doesn’t allow you to set anything, even though their docs say you can. However, I think a config option sounds like a great idea, especially if Flickr fixes their system then both would work, and you have the option to choose which permission set you want via both. It’s weird they don’t let you specify that as scopes like most.

**Comment by atymic** [comment]
Sure, could you add the config var to the PR and set it to null by default? Thanks!

**Comment by brianwozeniak** [comment]
Okay this should be good to go now using an `env` variable to be consistent with how everything is done in this repository. Let me know if you see any issues.

File: src/Flickr/Server.php [hunk 1]
```diff
@@ -22,7 +22,9 @@ public function urlTemporaryCredentials()
      */
     public function urlAuthorization()
     {
-        return 'https://www.flickr.com/services/oauth/authorize';
+        $perms = env('FLICKR_AUTHORIZATION_PERMS');
```
Comment (atymic) [inline src/Flickr/Server.php:25]: Please use `additionalConfigKeys` and `getConfig` and update the readme

Comment (brianwozeniak) [reply src/Flickr/Server.php:25]: Unfortunately, I do not see any documentation on utilizing this:

https://socialiteproviders.com/contribute/

Perhaps I missed where this is?

Additionally, I looked through some other providers using these methods and am failing to see documentation there that might provide some clues on exactly what you are wanting. I simply looked through numerous of the providers here and saw that `env` was used directly, so followed by example.

**Comment by atymic** [comment]
Sorry, could be documented better. We shouldn't use `env()` as it breaks config caching. I'll create an issue for it.

See auth0 as an example. To add keys:
https://github.com/SocialiteProviders/Providers/blob/e4a979bc081876bbcc37f74fa32ae3657bb31c00/src/Auth0/Provider.php#L35-L38

To use:
https://github.com/SocialiteProviders/Providers/blob/e4a979bc081876bbcc37f74fa32ae3657bb31c00/src/Auth0/Provider.php#L29

**Comment by lucasmichot** [comment]
I agree that some documentation is lacking.
@brianwozeniak feel free to merge this PR: https://github.com/brianwozeniak/Providers/pull/1

**Comment by lucasmichot** [comment]
The PR has been updated @atymic

**Comment by brianwozeniak** [comment]
Thanks @lucasmichot, merged yours in. @atymic I believe this is ready to go now, please check.

**Comment by atymic** [comment]
Actually, could you also add to the readme please? New PR is fine, then i'll tag new vers.

**Pull request merged**